rfc/2024-11-13-provider-oci-registry.md

```diff
@@ -0,0 +1,240 @@
+
+- The tag listing endpoint is located at `/v2/<name>/tags/list` and supports additional filtering and pagination parameters. It lists all the tags (a kind of reference) that have manifests.
+- The referrer listing endpoint is located at `/v2/<name>/referrers/<digest>` and returns the list of manifests that refer to a specific blob.
+
+> [!TIP]
+> Try it yourself: authenticate with a token as described above, then use the following command:
+> ```
+> curl -H "Authorization: Bearer djE6b3Blb..." https://ghcr.io/v2/opentofu/opentofu/tags/list
+> ```
+
+### The `_catalog` extension
+
+Although not standardized in the distribution spec, the `/v2/_catalog` endpoint is traditionally used in the [Docker Registry](https://docker-docs.uclv.cu/registry/spec/api/#listing-repositories) as a means to list all images in a registry. However, this endpoint is typically disabled in public registries or only available after authentication.
+
+---
+
+## User Documentation
+
+### For provider authors / mirror operators
+
+There are two possible approaches for distributing providers.
+
+<details>
+<summary>
+
+#### 📂 Standard container tooling approach
+
+</summary>
+
+As a provider author, you may want to package up your provider binary into an OCI image. This involves the following steps:
+
+1. Take all build artifacts and add them to a container image.
+2. Upload the container image to an OCI registry.
+
+For example, assuming you have your binary built and wanted to package it up into a container, you could use the following `Dockerfile`:
+
+```Dockerfile
+FROM scratch
+
+ADD terraform-provider-aws
+ADD README.md
+ADD LICENSE
+ADD CHANGELOG.md
+```
+
+You could now run `docker build -t ghcr.io/opentofu/terraform-provider-aws:1.2.3-amd64 .` followed by `docker push ghcr.io/opentofu/terraform-provider-aws:1.2.3`.
+
+For more advanced use cases, you could enable container image building in your [goreleaser config](https://goreleaser.com/customization/docker/) to enable building multi-arch images.
+</details>
+
+<details>
+<summary>
+
+#### 📂 ORAS approach
+
+</summary>
+
+As a provider author, you may want to package up your provider binary into an OCI image. You can do this by using the [ORAS command line tool](https://oras.land/docs/installation) using the following commands:
+
+```
+oras login -u username -p password ghcr.io
+oras push ghcr.io/opentofu/terraform-provider-aws:1.2.3 ./dist:application/octet-stream+tar
+```
+
+🛑🛑🛑 TODO: how do you create multi-arch with ORAS? 🛑🛑🛑
+
+</details>
+
+### Mirror configuration
+
+If you would like to use an OCI registry as your OpenTofu mirror, you will be able to do so by specifying a `provider_installation` block with an `oci_mirror` section as follows:
+
+```hcl
+provider_installation {
+  oci_mirror {
+    url = "ghcr.io/{{ .Namespace }}/terraform-provider-{{ .Name }}"
```

Comment:
Do we already have similar features that support template strings in any format?

If we don't, this syntax is something we would use for other similar features in the future so I would suggest considering other options as well.